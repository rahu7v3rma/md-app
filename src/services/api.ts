import axios, { AxiosError } from 'axios';
import moment from 'moment';

import { LogActivityRequestBody } from '@/types/log';
import {
    AUTHORIZATION_HEADER_NAME,
    getAuthorizationHeaderValue,
    setAuthToken
} from '@/utils/auth';
import { COMMON } from '@/utils/common';

const API_END_POINT = {
    LOGIN: 'user/login',
    GET_USER_JOURNEY: 'content/journey',
    GET_LESSON: 'content/lesson/{0}/content',
    COMPLETE_LESSON: 'content/lesson/{0}/complete',
    FIRST_CHANGE_PASSWORD: 'user/login/change-password',
    CHANGE_PASSWORD: 'user/change-password',
    GET_PROFILE: 'profile/',
    REFRESH_PROFILE_SESSION: 'profile/session',
    UPDATE_PROFILE: 'profile/',
    IMAGE_UPLOAD: 'image/upload',
    RESET_PASSWORD: 'user/reset/request',
    RESET_PASSWORD_VERIFY: 'user/reset/verify',
    RESET_PASSWORD_CONFIRM: 'user/reset/confirm',
    DELETE_ACCOUNT: 'user/delete',
    LOG_MEAL: 'log/food',
    LOG_EXERCISE: 'log/exercise',
    LOG_MEDICATION: 'log/medication',
    GET_LOG_PICKER_VALUES: 'log/picker-values',
    LOG_FAST: 'log/fast',
    UPDATE_FAST: 'log/fast/{0}/',
    LOG_WEIGHT: 'log/weight',
    LOG_WATER_INTAKE: 'log/drink',
    LOG_BLOOD: 'log/glucose',
    LOG_INSULIN: 'log/insulin',
    LOG_RECENT: 'log/recent?page={0}&limit={1}',
    UPDATE_LOG_MEDICATION: 'log/medication/{0}/',
    UPDATE_LOG_WEIGHT: 'log/weight/{0}/',
    UPDATE_LOG_BLOOD: 'log/glucose/{0}/',
    UPDATE_LOG_INSULIN: 'log/insulin/{0}/',
    UPDATE_LOG_WATER_INTAKE: 'log/drink/{0}/',
    UPDATE_LOG_MEAL: 'log/food/{0}/',
    UPDATE_LOG_ACTIVITY: 'log/exercise/{0}/',
    GET_DAILY_TASKS: 'content/daily-tasks',
    GET_COACH: 'user/coach',
    DAILY_COMPLETED_LOGS: '/log/daily-completed-logs-task?date={0}',
    DELETE_MEAL_LOG: '/log/food/delete/{0}/',
    DELETE_WATER_IN_TAKE_LOG: '/log/drink/delete/{0}/',
    DELETE_BLOOD_LOG: '/log/glucose/delete/{0}/',
    DELETE_FAST_LOG: 'log/fast/delete/{0}/',
    DELETE_LOG_WEIGHT: '/log/weight/delete/{0}/',
    DELETE_ACTIVITY_LOG: 'log/exercise/delete/{0}/',
    DELETE_LOG_INSULIN: 'log/insulin/delete/{0}/',
    DELETE_LOG_MEDICATION: '/log/medication/delete/{0}/',
    NOTIFICATION_LIST: 'user/notification?page={0}&limit={1}',
    UPDATE_NOTIFICATION: 'user/notification/{0}',
    GET_CLIENT_CONFIG: 'config/{0}/{1}'
};

type API_METHOD = 'GET' | 'POST' | 'PUT' | 'DELETE' | 'PATCH';

export const login = async (email: string, password: string) => {
    return _baseRequest(API_END_POINT.LOGIN, 'POST', { email, password }).then(
        (loginResponse) => {
            const authToken = loginResponse.auth_token;
            delete loginResponse.auth_token;

            setAuthToken(authToken);

            return loginResponse;
        }
    );
};

export const firstChangePassword = async (
    email: string,
    old_password: string,
    new_password: string
) => {
    return _baseRequest(API_END_POINT.FIRST_CHANGE_PASSWORD, 'POST', {
        email,
        old_password,
        new_password
    }).then((loginResponse) => {
        const authToken = loginResponse.auth_token;
        delete loginResponse.auth_token;
        setAuthToken(authToken);

        return loginResponse;
    });
};

export const changePassword = async (
    old_password: string,
    new_password: string
) => {
    return _authorizedRequest(API_END_POINT.CHANGE_PASSWORD, 'POST', {
        old_password,
        new_password
    }).then((loginResponse) => {
        const authToken = loginResponse.new_token;
        delete loginResponse.new_token;
        setAuthToken(authToken);

        return loginResponse;
    });
};

export const resetPassword = async (email: string, client: string) => {
    return await _baseRequest(API_END_POINT.RESET_PASSWORD, 'POST', {
        email,
        client
    });
};

export const resetPasswordVerify = async (code: string) => {
    return await _baseRequest(API_END_POINT.RESET_PASSWORD_VERIFY, 'POST', {
        token: code
    });
};

export const resetPasswordConfirm = async (code: string, password: string) => {
    return await _baseRequest(API_END_POINT.RESET_PASSWORD_CONFIRM, 'POST', {
        token: code,
        password
    });
};

export const getClientConfig = async (platform: string, version: string) => {
    return await _baseRequest(
        COMMON.stringFormat(API_END_POINT.GET_CLIENT_CONFIG, platform, version)
    );
};

export const deleteAccount = async (password: string) => {
    return await _authorizedRequest(API_END_POINT.DELETE_ACCOUNT, 'POST', {
        password
    });
};

export const getUserJourney = async () => {
    return await _authorizedRequest(API_END_POINT.GET_USER_JOURNEY);
};

export const getLesson = async (lessonId: number) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.GET_LESSON, lessonId)
    );
};

export const completeLesson = async (lessonId: number) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.COMPLETE_LESSON, lessonId),
        'PUT',
        {
            completion_time: moment().format('YYYY-MM-DD HH:mm:ss')
        }
    );
};

export const createLogMeal = async (logTime: string, image: string) => {
    return await _authorizedRequest(API_END_POINT.LOG_MEAL, 'POST', {
        log_time: logTime,
        image
    });
};

export const updateLogMeal = async (
    id: number | string,
    logTime: string,
    image: string
) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.UPDATE_LOG_MEAL, id),
        'PUT',
        {
            log_time: logTime,
            image
        }
    );
};

export const createLogWeight = async (
    logTime: string,
    amount: number,
    unit: string
) => {
    return await _authorizedRequest(API_END_POINT.LOG_WEIGHT, 'POST', {
        log_time: logTime,
        amount: amount,
        unit: unit
    });
};

export const createLogFast = async (
    logTime: string,
    durationMinutes: number
) => {
    return await _authorizedRequest(API_END_POINT.LOG_FAST, 'POST', {
        log_time: logTime,
        duration_minutes: durationMinutes
    });
};

export const updateLogActivity = async (
    id: number | string,
    activity_type: string,
    duration_minutes: number | string,
    intensity: number | string,
    log_time: string
) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.UPDATE_LOG_ACTIVITY, id),
        'PUT',
        {
            activity_type: activity_type,
            duration_minutes: duration_minutes,
            intensity: intensity,
            log_time: log_time
        }
    );
};
export const updateLogFast = async (
    id: number,
    logTime: string,
    durationMinutes: number
) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.UPDATE_FAST, id),
        'PUT',
        {
            log_time: logTime,
            duration_minutes: durationMinutes
        }
    );
};

export const createWaterIntakeLog = async (
    log_time: string,
    amount: number,
    unit: string
) => {
    return await _authorizedRequest(API_END_POINT.LOG_WATER_INTAKE, 'POST', {
        log_time,
        amount,
        unit
    });
};

export const updateWaterIntakeLog = async (
    id: number | string,
    log_time: string,
    amount: number,
    unit: string
) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.UPDATE_LOG_WATER_INTAKE, id),
        'PUT',
        {
            log_time,
            amount,
            unit
        }
    );
};

export const createLogBlood = async (
    logTime: string,
    amount: number,
    unit: string,
    measurementType: string
) => {
    return await _authorizedRequest(API_END_POINT.LOG_BLOOD, 'POST', {
        log_time: logTime,
        amount,
        unit,
        measurement_type: measurementType
    });
};

export const updateLogBlood = async (
    id: number | string,
    logTime: string,
    amount: number,
    unit: string,
    measurementType: string
) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.UPDATE_LOG_BLOOD, id),
        'PUT',
        {
            log_time: logTime,
            amount,
            unit,
            measurement_type: measurementType
        }
    );
};

export const createLogInsulin = async (
    logTime: string,
    units: number,
    injectionType: string
) => {
    return await _authorizedRequest(API_END_POINT.LOG_INSULIN, 'POST', {
        log_time: logTime,
        units,
        injection_type: injectionType
    });
};

export const updateLogInsulin = async (
    id: number | string,
    logTime: string,
    units: number,
    injectionType: string
) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.UPDATE_LOG_INSULIN, id),
        'PUT',
        {
            log_time: logTime,
            units,
            injection_type: injectionType
        }
    );
};

export const deleteLogInsulin = async (id: number | string) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.DELETE_LOG_INSULIN, id),
        'DELETE'
    );
};

export const getProfile = async () => {
    return await _authorizedRequest(API_END_POINT.GET_PROFILE);
};

export const refreshProfileSession = async (
    fcmToken?: string,
    deviceId?: string,
    deviceType?: string,
    utcTimeDiffHours?: number
) => {
    return await _authorizedRequest(
        API_END_POINT.REFRESH_PROFILE_SESSION,
        'POST',
        {
            fcm_token: fcmToken,
            device_id: deviceId,
            device_type: deviceType,
            utc_time_diff_hours: utcTimeDiffHours
        }
    );
};

export const updateProfile = async (image: string) => {
    return await _authorizedRequest(API_END_POINT.UPDATE_PROFILE, 'PUT', {
        image
    });
};

export const getImageUploadCredentials = async (publicImage: boolean) => {
    let apiPath = API_END_POINT.IMAGE_UPLOAD;
    if (publicImage) {
        apiPath = apiPath.concat('?public=1');
    }

    return await _authorizedRequest(apiPath);
};

export const logExercise = async (object: LogActivityRequestBody) => {
    return await _authorizedRequest(API_END_POINT.LOG_EXERCISE, 'POST', object);
};

export const logMedication = async (
    logTime: string,
    amount: number,
    drugName: string,
    dose: string
) => {
    return await _authorizedRequest(API_END_POINT.LOG_MEDICATION, 'POST', {
        log_time: logTime,
        amount: amount,
        drug_name: drugName,
        dose: dose
    });
};

export const updateLogMedication = async (
    id: string,
    logTime: string,
    amount: number,
    drugName: string,
    dose: string
) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.UPDATE_LOG_MEDICATION, id),
        'PUT',
        {
            log_time: logTime,
            amount: amount,
            drug_name: drugName,
            dose: dose
        }
    );
};

export const deleteLogMedication = async (id: string) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.DELETE_LOG_MEDICATION, id),
        'DELETE'
    );
};

export const getLogPickerValues = async () => {
    return await _authorizedRequest(API_END_POINT.GET_LOG_PICKER_VALUES);
};

export const getRecentLog = async (page: number, limit: number) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.LOG_RECENT, page, limit)
    );
};

export const getNotificationList = async (page: number, limit: number) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.NOTIFICATION_LIST, page, limit)
    );
};

export const updateNotification = async (id: number) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.UPDATE_NOTIFICATION, id),
        'PATCH'
    );
};

export const updateLogWeight = async (
    logTime: string,
    amount: number,
    unit: string,
    logId: number
) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.UPDATE_LOG_WEIGHT, logId),
        'PUT',
        {
            log_time: logTime,
            amount: amount,
            unit: unit
        }
    );
};

export const getDailyTasks = async () => {
    return await _authorizedRequest(API_END_POINT.GET_DAILY_TASKS);
};

export const getCoach = async () =>
    await _authorizedRequest(API_END_POINT.GET_COACH);

export const getDailyCompletedLogs = async (date: Date) => {
    return await _authorizedRequest(
        COMMON.stringFormat(
            API_END_POINT.DAILY_COMPLETED_LOGS,
            moment(date).format('YYYY-MM-DD')
        )
    );
};

export const deleteActivityLog = async (logId: string) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.DELETE_ACTIVITY_LOG, logId),
        'DELETE'
    );
};

export const deleteMealLog = async (logId: string) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.DELETE_MEAL_LOG, logId),
        'DELETE'
    );
};

export const deleteWaterInTakeLog = async (logId: string) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.DELETE_WATER_IN_TAKE_LOG, logId),
        'DELETE'
    );
};

export const deleteBloodLog = async (logId: string) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.DELETE_BLOOD_LOG, logId),
        'DELETE'
    );
};

export const deleteLogWeight = async (logId: string) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.DELETE_LOG_WEIGHT, logId),
        'DELETE'
    );
};

const _baseRequest = (
    url: string,
    method: API_METHOD = 'GET',
    data?: object,
    headers?: { [key: string]: string }
): Promise<any> => {
    return apiClient
        .request({
            method,
            url,
            headers,
            data
        })
        .then((response) => response.data.data)

        .catch((err: AxiosError | Error) =>
            Promise.reject({
                name: err.name,
                message: err.message,
                status: (err as AxiosError).response?.status || -1,
                data: (err as AxiosError).response?.data
            })
        );
};

export const deleteLogFast = async (logId: string) => {
    return await _authorizedRequest(
        COMMON.stringFormat(API_END_POINT.DELETE_FAST_LOG, logId),
        'DELETE'
    );
};

const _authorizedRequest = (
    url: string,
    method: API_METHOD = 'GET',
    data?: object,
    headers?: { [key: string]: string }
): Promise<any> => {
    return getAuthorizationHeaderValue().then((authHeaderValue) => {
        if (!headers) {
            headers = {};
        }
        headers[AUTHORIZATION_HEADER_NAME] = authHeaderValue || '';
        return _baseRequest(url, method, data, headers);
    });
};

const apiClient = axios.create({
    baseURL: COMMON.apiBaseUrl,
    headers: {
        'Content-type': 'application/json'
    }
});
