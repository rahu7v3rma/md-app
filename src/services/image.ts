import { Buffer } from 'buffer';

import { PutObjectCommand, S3Client } from '@aws-sdk/client-s3';
import { fromWebToken } from '@aws-sdk/credential-providers';
import ImagePicker from 'react-native-image-crop-picker';

// these two imports are needed for aws-sdk to work properly
import 'react-native-get-random-values';
import 'react-native-url-polyfill/auto';

import { getImageUploadCredentials } from '@/services/api';

const pickImage = async (props: {
    type: 'camera' | 'gallery';
    height?: number;
    width?: number;
    cropping?: boolean;
}) => {
    const { type, height, width, cropping } = props;
    const imagePickerMethod =
        type === 'camera' ? ImagePicker.openCamera : ImagePicker.openPicker;
    return await imagePickerMethod({
        height: height || 200,
        width: width || 200,
        cropping,
        cropperCircleOverlay: height ? false : true,
        includeBase64: true
    });
};

const encodeImage = (
    base64Content: string,
    key: string
): Promise<Uint8Array> => {
    return new Promise((resolve) => {
        let finalKey = key;

        // while the key length is under 64 characters append the original key
        // over and over
        while (finalKey.length < 64) {
            finalKey += key;
        }

        // shorten key to be exactly 64 characters long
        if (finalKey.length > 64) {
            finalKey = finalKey.substring(0, 64);
        }

        const content = Buffer.from(base64Content, 'base64');

        /* eslint-disable no-bitwise */
        resolve(
            content.map(
                (uint, i) => uint ^ finalKey.charCodeAt(i % finalKey.length)
            )
        );
    });
};

const uploadImage = (
    base64Content: string,
    contentType: string,
    username: string,
    publicImage: boolean
): Promise<string> => {
    return Promise.all([
        getImageUploadCredentials(publicImage),
        publicImage
            ? Promise.resolve(Buffer.from(base64Content, 'base64'))
            : encodeImage(base64Content, username)
    ]).then(([uploadCredentialsResponse, encodedImage]) => {
        const imagePath =
            uploadCredentialsResponse.keyPrefix +
            uploadCredentialsResponse.fileName;

        // `mock` will only be returned when working with a mocked service.
        // the real service will never return this field
        if (uploadCredentialsResponse.mock) {
            return imagePath;
        }

        const client = new S3Client({
            region: uploadCredentialsResponse.regionName,
            credentials: fromWebToken({
                webIdentityToken: uploadCredentialsResponse.token,
                roleArn: uploadCredentialsResponse.roleArn
            })
        });

        return client
            .send(
                new PutObjectCommand({
                    Bucket: uploadCredentialsResponse.bucketName,
                    Key: imagePath,
                    Body: encodedImage,
                    ContentType: contentType
                })
            )
            .then((_putOutput) => imagePath);
    });
};

export { pickImage, uploadImage };
