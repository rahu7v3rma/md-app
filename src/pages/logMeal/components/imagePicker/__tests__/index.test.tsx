import React from 'react';
import renderer, { act } from 'react-test-renderer';

import imageMock from '@/pages/logMeal/__mocks__/image.json';
import ImagePicker from '@/pages/logMeal/components/imagePicker';
import * as imageServices from '@/services/image';

jest.spyOn(imageServices, 'pickImage').mockResolvedValue(imageMock);

it('imagePicker snapshot', async () => {
    const tree = renderer.create(<ImagePicker onImagePicked={() => {}} />);
    expect(tree).toMatchSnapshot();
});

it('imagePicker renders without errors', async () => {
    const tree = renderer.create(<ImagePicker onImagePicked={() => {}} />);
    expect(tree.root).toBeTruthy();
});

it('opens the image chooser when the image is pressed', async () => {
    const tree = renderer.create(
        <ImagePicker
            onImagePicked={() => {}}
            imageChooserTouchID={'imageChooserTouchID'}
        />
    );
    let imageChooserTouchID = tree.root.findByProps({
        testID: 'imageChooserTouchID'
    }).props;
    await act(() => imageChooserTouchID.onPress());
    let choosePhoto = tree.root.findByProps({
        testID: 'choosePhoto'
    }).props;
    expect(choosePhoto).toBeTruthy();
});

it('calls onImagePicked when an image is selected', async () => {
    jest.spyOn(console, 'warn').mockImplementation(() => {});
    const onImagePicked = jest.fn();
    const tree = renderer.create(
        <ImagePicker
            onImagePicked={onImagePicked}
            imageChooserTouchID={'imageChooserTouchID'}
        />
    );
    let imageChooserTouchID = tree.root.findByProps({
        testID: 'imageChooserTouchID'
    }).props;
    await act(() => imageChooserTouchID.onPress());
    let choosePhoto = tree.root.findByProps({
        testID: 'choosePhoto'
    }).props;
    await act(() => choosePhoto.onPress());
    expect(onImagePicked).toHaveBeenCalledTimes(1);
});

it('clears the selected image when delete button is pressed', async () => {
    jest.spyOn(console, 'warn').mockImplementation(() => {});
    const onImagePicked = jest.fn();
    const tree = renderer.create(
        <ImagePicker
            onImagePicked={onImagePicked}
            imageChooserTouchID={'imageChooserTouchID'}
            deleteImageID={'deleteImageID'}
        />
    );
    let imageChooserTouchID = tree.root.findByProps({
        testID: 'imageChooserTouchID'
    }).props;
    await act(() => imageChooserTouchID.onPress());
    let choosePhoto = tree.root.findByProps({
        testID: 'choosePhoto'
    }).props;
    await act(() => choosePhoto.onPress());
    expect(onImagePicked).toHaveBeenCalledTimes(1);
    let deleteImageID = tree.root.findByProps({
        testID: 'deleteImageID'
    }).props;
    await act(() => deleteImageID.onPress());
    let imagePickerButton = tree.root.findByProps({
        testID: 'imagePickerButton'
    }).props;
    expect(imagePickerButton).toBeTruthy();
});
