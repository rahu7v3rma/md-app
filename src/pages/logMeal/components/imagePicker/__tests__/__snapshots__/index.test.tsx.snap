// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imagePicker snapshot 1`] = `
<View
  style={
    {
      "paddingVertical": 10,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 12,
        "display": "flex",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderRadius": 12,
          "height": 209,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="imagePickerButton"
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={40}
          bbWidth={48}
          fill="none"
          focusable={false}
          height={40}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 40,
                "width": 48,
              },
            ]
          }
          vbHeight={40}
          vbWidth={48}
          width={48}
          xml="<svg width="48" height="40" viewBox="0 0 48 40" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path
        d="M31.7413 23.239C31.7413 27.5077 28.2687 30.9803 24 30.9803C19.7313 30.9803 16.2601 27.5077 16.2601 23.239C16.2601 18.9703 19.7313 15.4978 24 15.4978C28.2687 15.4978 31.7413 18.9716 31.7413 23.239ZM48 12.2468V34.2339C48 37.165 45.6235 39.5415 42.6923 39.5415H5.30768C2.37651 39.5415 0 37.165 0 34.2339V12.2468C0 9.31564 2.37651 6.93913 5.30768 6.93913H11.8361V5.10267C11.8361 2.53773 13.9141 0.45845 16.4803 0.45845H31.5197C34.0859 0.45845 36.1639 2.53773 36.1639 5.10267V6.9378H42.6923C45.6235 6.93913 48 9.31564 48 12.2468ZM35.722 23.239C35.722 16.7756 30.4634 11.517 24 11.517C17.5379 11.517 12.2793 16.7756 12.2793 23.239C12.2793 29.7024 17.5379 34.961 24 34.961C30.4634 34.961 35.722 29.7024 35.722 23.239Z"
        fill="#D3E29F" />
</svg>"
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M31.7413 23.239C31.7413 27.5077 28.2687 30.9803 24 30.9803C19.7313 30.9803 16.2601 27.5077 16.2601 23.239C16.2601 18.9703 19.7313 15.4978 24 15.4978C28.2687 15.4978 31.7413 18.9716 31.7413 23.239ZM48 12.2468V34.2339C48 37.165 45.6235 39.5415 42.6923 39.5415H5.30768C2.37651 39.5415 0 37.165 0 34.2339V12.2468C0 9.31564 2.37651 6.93913 5.30768 6.93913H11.8361V5.10267C11.8361 2.53773 13.9141 0.45845 16.4803 0.45845H31.5197C34.0859 0.45845 36.1639 2.53773 36.1639 5.10267V6.9378H42.6923C45.6235 6.93913 48 9.31564 48 12.2468ZM35.722 23.239C35.722 16.7756 30.4634 11.517 24 11.517C17.5379 11.517 12.2793 16.7756 12.2793 23.239C12.2793 29.7024 17.5379 34.961 24 34.961C30.4634 34.961 35.722 29.7024 35.722 23.239Z"
              fill={4292076191}
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#D3E29F40",
      }
    }
  >
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      supportedOrientations={
        [
          "portrait",
          "portrait-upside-down",
          "landscape",
          "landscape-left",
          "landscape-right",
        ]
      }
      transparent={true}
      visible={false}
    />
  </View>
</View>
`;
